task4/task4.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+timestamp=$(date +"%H:%M:%S")
+new_ps1="\[${timestamp}\] >"
+
+current_ps1="$PS1"
+
+PS1="$new_ps1"
+
+dotfile_path="$HOME/.dotfile"
+if [ -e "$dotfile_path" ]; then
+  source "$dotfile_path"
+fi
+
+# Display a message with the updated PS1
+echo "Temporary PS1 set to: $PS1"
+
+# Restore the original PS1
+PS1="$current_ps1"
```

Comment:
We need to run this .dotfile from ~/.bashrc file. So, you'll have to add your PS1 (written in your dotfile dotfile) into your ~/.bashrc file. In simple write your ps1 in .dotfle and append the contents of that dotfile into the ~/.basrhc file.
I hope you'll get that.